code/Helper/Entity/Producthelper.php

```diff
@@ -817,7 +817,7 @@ public function getObject(Mage_Catalog_Model_Product $product)
                         // sub-products is out of stock.
                         $customData['in_stock'] = 0;
                     }
-                } else {
+                } elseif ($value === null) {
                     $value = $this->getValueOrValueText($product, $attribute_name, $attribute_resource);
```

Comment:
Seems ok, but we need an else no ? Otherwise the value won't be set in the customData variable